functional-test/src/main/java/org/zanata/page/AbstractPage.java

```diff
@@ -435,4 +371,203 @@ private void assertReady(WebElement targetElement) {
+                log.info("Missed a notification X click");
+            }
+        }
+        // Finally, forcibly un-is-active the message container - for speed
+        String script = "return (typeof $ == 'undefined') ?  [] : " +
+                "$('ul.message--global').toArray()";
+        @SuppressWarnings("unchecked")
+        List<WebElement> messageBoxes = ((List<WebElement>) getExecutor()
+                .executeScript(script));
+        for (WebElement messageBox : messageBoxes) {
+            getExecutor().executeScript(
+                    "arguments[0].setAttribute('class', arguments[1]);",
+                    messageBox,
+                    messageBox.getAttribute("class").replace("is-active", ""));
+        }
+    }
+
+    /**
+     * Wait for the notifications box to go. Assumes test has dealt with
+     * removing it, or is waiting for it to time out.
+     */
+    public void waitForNotificationsGone() {
+        final String script = "return (typeof $ == 'undefined') ?  [] : " +
+                "$('ul.message--global').toArray()";
+        final String message = "Waiting for notifications box to go";
+        waitForAMoment().withMessage(message).until(new Predicate<WebDriver>() {
+            @Override
+            public boolean apply(WebDriver input) {
+                @SuppressWarnings("unchecked")
+                List<WebElement> boxes = (List<WebElement>) getExecutor()
+                        .executeScript(script);
+                for (WebElement box : boxes) {
+                    if (box.isDisplayed()) {
+                        log.info(message);
+                        return false;
+                    }
+                }
+                return true;
+            }
+        });
+    }
+
+    /**
+     * Shift focus to the page, in order to activate some elements that only
+     * exhibit behaviour on losing focus. Some pages with contained objects
+     * cause object behaviour to occur when interacted with, so in this case
+     * interact with the container instead.
+     */
+    public void defocus() {
+        log.info("Click off element focus");
+        List<WebElement> webElements =
+                getDriver().findElements(By.id("container"));
+        webElements.addAll(getDriver().findElements(By.tagName("body")));
+        if (webElements.size() > 0) {
+            webElements.get(0).click();
+        } else {
+            log.warn("Unable to focus page container");
+        }
+    }
+
+    /**
+     * Force the blur 'unfocus' process on a given element
+     */
+    public void defocus(By elementBy) {
+        log.info("Force unfocus");
+        WebElement element = existingElement(elementBy);
+        getExecutor().executeScript("arguments[0].blur()", element);
+        waitForPageSilence();
+    }
+
+    /* The system sometimes moves too fast for the Ajax pages, so provide a
+     * pause
+     */
+    public void slightPause() {
+        try {
+            Thread.sleep(500);
```

Comment:
Yes, an old artifact of dodgy development. It's not a new addition, but moved from elsewhere. The sooner I can remove it entirely, the better.